var/spack/repos/builtin/packages/miopen-hip/package.py

```diff
@@ -25,12 +25,14 @@ class MiopenHip(CMakePackage):
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
 
     depends_on('cmake@3:', type='build')
-    depends_on('boost@1.58.0', type='link')
+    depends_on('boost@1.67.0:1.73.0', type='link')
     depends_on('pkgconfig', type='build')
     depends_on('bzip2', type='link')
     depends_on('sqlite', type='link')
     depends_on('half', type='build')
     depends_on('zlib', type='link', when='@3.9.0:')
+    conflicts('^boost@1.74.0')
+    conflicts('^boost@1.75.0')
```

Comment:
These two conflicts aren't needed if the dependency if limited to a range finishing at 1.73.0